### PR TITLE
Cdd 1585 v3

### PR DESCRIPTION
**Pull request by cichockimc**

### CODE REVIEW

File: test/util/externalservices/MdgWcoDecService.scala [hunk 1]
```diff
@@ -17,24 +17,46 @@
...
 trait MdgWcoDecService extends WireMockRunner {
   private val urlV1MatchingRequestPath = urlMatching(CustomsDeclarationsExternalServicesConfig.MdgWcoDecV1ServiceContext)
   private val urlV2MatchingRequestPath = urlMatching(CustomsDeclarationsExternalServicesConfig.MdgWcoDecV2ServiceContext)
+  private val urlV3MatchingRequestPath = urlMatching(CustomsDeclarationsExternalServicesConfig.MdgWcoDecV3ServiceContext)
 
-  def startMdgWcoDecService(): Unit = {
-    setupMdgWcoDecServiceToReturn(ACCEPTED)
+
+  def startMdgWcoDecServiceV1(): Unit = {
+    setupMdgWcoDecServiceToReturn(ACCEPTED, urlV2MatchingRequestPath)
```
Comment (paulblakehmrc) [inline test/util/externalservices/MdgWcoDecService.scala]: Need to check why these are passing as v1 and v3 are matching against the wrong path. v2 is ok.

File: test/unit/connectors/DeclarationConnectorSpec.scala [hunk 2]
```diff
@@ -168,9 +170,14 @@ class DeclarationConnectorSpec extends UnitSpec with MockitoSugar with BeforeAnd
         headersCaptor.getValue.extraHeaders should contain("X-Correlation-ID" -> correlationId.toString)
       }
 
+      //TODO MC v3
```
Comment (paulblakehmrc) [inline test/unit/connectors/DeclarationConnectorSpec.scala]: Looks like this TODO isn't needed now

File: test/component/DeclarationSchemaSpec.scala [hunk 3]
```diff
@@ -43,7 +43,7 @@ class DeclarationSchemaSpec extends ComponentTestSpec
     startMockServer()
     stubAuditService()
     authServiceAuthorizesCSP()
-    startMdgWcoDecService()
+    startMdgWcoDecServiceV2()
```
Comment (mattclark-zerogravit) [inline test/component/DeclarationSchemaSpec.scala]: We are only testing V2 schema here do we want to test v1 & v3 as well?

File: test/unit/connectors/DeclarationConnectorSpec.scala (hunk 2 above)
Comment (mattclark-zerogravit) [inline test/unit/connectors/DeclarationConnectorSpec.scala]: Please remove todo

Comment (mattclark-zerogravit) [inline test/unit/connectors/DeclarationConnectorSpec.scala]: remove todo

File: test/unit/controllers/actionbuilders/HeaderValidatorSpec.scala [hunk 4]
```diff
@@ -31,6 +31,7 @@ import uk.gov.hmrc.play.test.UnitSpec
 import util.RequestHeaders.{ValidHeadersV2, _}
 import util.{ApiSubscriptionFieldsTestData, TestData}
 
+//TODO MC add v3
```
Comment (mattclark-zerogravit) [inline test/unit/controllers/actionbuilders/HeaderValidatorSpec.scala]: Looks like this is missing V3 tests

Comment (cichockimc) [reply test/unit/controllers/actionbuilders/HeaderValidatorSpec.scala]: please note `//TODO: Confirm with QA's that schema tests like these that give confidence around schema changes are tested manually by QA's`
I would leave it as is.

File: test/component/DeclarationSchemaSpec.scala (hunk 3 above)
Comment (cichockimc) [reply test/component/DeclarationSchemaSpec.scala]: please note `//TODO: Confirm with QA's that schema tests like these that give confidence around schema changes are tested manually by QA's`
I would leave it as is.

**Comment by paulblakehmrc** [comment]
Perhaps a v3 section in application.conf?

**Pull request merged**